src/Build.cmd

```diff
@@ -2,6 +2,38 @@
 setlocal
 
 SET CMDHOME=%~dp0.
+
+@REM Locate VS 2017 with the proper method
+
+SET VSWHERE_REMOTE_PATH=https://github.com/Microsoft/vswhere/releases/download/1.0.55/vswhere.exe
+SET VSWHERE_LOCAL_PATH=%CMDHOME%\vswhere.exe
+
+if NOT exist "%VSWHERE_LOCAL_PATH%" (
+  echo Downloading vswhere.exe from %VSWHERE_REMOTE_PATH%.
+  powershell -NoProfile -ExecutionPolicy unrestricted -Command "$retryCount = 0; $success = $false; do { try { (New-Object Net.WebClient).DownloadFile('%VSWHERE_REMOTE_PATH%', '%VSWHERE_LOCAL_PATH%'); $success = $true; } catch { if ($retryCount -ge 6) { throw; } else { $retryCount++; Start-Sleep -Seconds (5 * $retryCount); } } } while ($success -eq $false);"
+  if NOT exist "%VSWHERE_LOCAL_PATH%" (
+    echo ERROR: Could not install vswhere correctly.
+    exit /b 1
+  )
+)
+
+set pre=Microsoft.VisualStudio.Product.
+set ids=%pre%Community %pre%Professional %pre%Enterprise %pre%BuildTools
+for /f "usebackq tokens=1* delims=: " %%i in (`%VSWHERE_LOCAL_PATH% -latest -products %ids% -requires Microsoft.Component.MSBuild`) do (
+  if /i "%%i"=="installationPath" set VS2017InstallDir=%%j
+)
+
+if NOT "%VS2017InstallDir%" == "" (
```

Comment:
btw, you can use this syntax that is a little bit more readable:
` if defined VS2017InstallDir`